src/Magento/FunctionalTestingFramework/Extension/TestContextExtension.php

```diff
@@ -150,6 +155,30 @@ public function extractContext($trace, $class)
         return null;
     }
 
+    /**
+     * Attach suppressed exception thrown before _after hook to the current step.
+     * @param  \Exception $exception
+     * @return mixed
+     */
+    public function logPreviousException(\Exception $exception)
+    {
+        $change = function () {
+            if ($this instanceof \PHPUnit\Framework\ExceptionWrapper) {
+                return $this->previous;
+            } else {
+                return $this->getPrevious();
+            }
+        };
+        $firstException = $change->call($exception);
+        $bind = function () use ($firstException) {
```

Comment:
I'm not sure what 173~176 do. `ExceptionWrapper.php` does not have a variable named `$exception`, and at this point you already have the thing you need to log.

Is this leftover test flow or something like that?

Reply:
removed the redundant code.